### PR TITLE
deps: promote tokio to workspace dependency

### PR DESCRIPTION
**Pull request by drHuangMHT**
## Description

Promote `tokio` to workspace dependency.  
Related issue: #5230

## Notes & open questions

Not sure if setting `default-features` to `false` on workspace level causes any problem.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by drHuangMHT** [comment]
@jxs Ready for review.

**Comment by drHuangMHT** [comment]
Ping @jxs got a minute to review this?

File: transports/dns/src/lib.rs [hunk 1]
```diff
@@ -777,7 +777,7 @@ mod tests {
             // type record lookups may not work with the system DNS resolver.
             let config = ResolverConfig::quad9();
             let opts = ResolverOpts::default();
-            let rt = tokio_crate::runtime::Builder::new_current_thread()
+            let rt = ::tokio::runtime::Builder::new_current_thread()
```
Comment (jxs) [inline transports/dns/src/lib.rs:780]: nit:
```suggestion
            let rt = tokio::runtime::Builder::new_current_thread()
```
should work no?

**Review by jxs — APPROVED** [review]

**Comment by jxs** [comment]
Thanks for this! LGTM only one question

File: transports/dns/src/lib.rs (hunk 1 above)
Comment (drHuangMHT) [reply transports/dns/src/lib.rs:780]: No because `super::*` is imported, and it contains module `tokio`.

Comment (jxs) [reply transports/dns/src/lib.rs:780]: thanks for clarifying

**Pull request merged**